dev/com.ibm.ws.microprofile.openapi_fat/fat/src/com/ibm/ws/microprofile/openapi/fat/ProxySupportTest.java

```diff
@@ -0,0 +1,135 @@
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.microprofile.openapi.fat;
+
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.ibm.websphere.simplicity.ShrinkHelper;
+import com.ibm.ws.microprofile.openapi.fat.utils.OpenAPIConnection;
+import com.ibm.ws.microprofile.openapi.fat.utils.OpenAPITestUtil;
+
+import componenttest.annotation.Server;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.utils.FATServletClient;
+import componenttest.topology.utils.HttpUtils;
+import test.common.TestLogger;
+
+/**
+ * Tests to ensure that the proxy support feature changes server object in the final document.
+ * This class tests the following scenarios: requests to /openapi with no Referer header, requests
+ * to /openapi with Referer header but different ports than the server and finally requests to /openapi
+ * with Referer header and same ports than the server. In these three scenarios, requests are sent
+ * to both http and https ports.
+ */
+@SuppressWarnings("restriction")
+@RunWith(FATRunner.class)
+public class ProxySupportTest extends FATServletClient {
+    private static final Class<?> c = ProxySupportTest.class;
+    private static final String REFERER = "Referer";
+
+    private static final String APP_NAME_1 = "pure-jaxrs";
+
+    @Server("ApplicationProcessorServer")
+    public static LibertyServer server;
+
+    @Rule
+    public final TestLogger logger = new TestLogger();
+
+    @BeforeClass
+    public static void setUpTest() throws Exception {
+        HttpUtils.trustAllCertificates();
+
+        ShrinkHelper.defaultApp(server, APP_NAME_1, "app.web.pure.jaxrs");
+
+        LibertyServer.setValidateApps(false);
+        server.startServer(c.getSimpleName() + ".log");
+
+        OpenAPITestUtil.addApplication(server, APP_NAME_1);
+        OpenAPITestUtil.waitForApplicationProcessorProcessedEvent(server, APP_NAME_1);
+        OpenAPITestUtil.waitForApplicationProcessorAddedEvent(server, APP_NAME_1);
+    }
+
+    /**
+     * This ensures all the applications are removed before running each test to make sure
+     * we start with a clean server.xml.
+     */
+    @Before
+    public void setUp() throws Exception {
+        // Change server ports to the default ones
+        OpenAPITestUtil.changeServerPorts(server, server.getHttpDefaultPort(), server.getHttpDefaultSecurePort());
+    }
+
+    @AfterClass
+    public static void tearDown() throws Exception {
+        server.stopServer("CWWKO1650E");
```

Comment:
is the test using invalid app(s) to ignore validation errors?

Reply:
See this [comment](https://github.com/arturdzm/open-liberty/pull/64#discussion_r168793895).